cognite/src/api/data_modeling/resource.rs

```diff
@@ -0,0 +1,155 @@
+use std::future::Future;
+use std::sync::Arc;
+
+use serde::de::DeserializeOwned;
+use serde::Serialize;
+
+use crate::models::data_models::{FromReadable, IntoWritable};
+use crate::models::instances::{
+    NodeAndEdgeCreateCollection, NodeAndEdgeRetrieveRequest, NodeAndEdgeRetrieveResponse,
+    NodeOrEdgeCreate, NodeOrEdgeSpecification, SlimNodeOrEdge, SourceReferenceInternal,
+};
+use crate::models::views::ViewReference;
+use crate::{Result, RetrieveWithRequest, UpsertCollection};
+
+use super::instances::Instances;
+/// Data model files.
+pub mod files;
+
+/// Trait for data models special instance
+pub trait WithView {
+    /// Default space
+    const SPACE: &'static str;
+    /// Default external ID
+    const EXTERNAL_ID: &'static str;
+    /// Default version
+    const VERSION: &'static str;
+
+    /// Set custom view for instance.
+    ///
+    /// # Arguments
+    ///
+    /// * `space` - View space.
+    /// * `external_id` - View external id.
+    /// * `version` - View version.
+    fn with_view(&mut self, space: String, external_id: String, version: String);
+    /// Get view for instance
+    fn view(&self) -> ViewReference;
```

Comment:
```suggestion
    fn view(&self) -> &ViewReference;
```

There's no reason to require a clone here. Instead, the clone can be done by the caller, which is more general.